### ISSUE TITLE
Kubernetes 1.12 pod security policies are causing errors when installing weave agents.

### ISSUE THREAD

**Issue by bricef** [weaveworks/launcher#261]
Weave cloud agent config https://cloud.weave.works/k8s.yaml?k8s-version=v1.12.1&t=rfc9saahqrj9xnap6cyyyjsqjxp9e4mj&omit-support-info=true fails to install with the following agent error:

```
time="2018-10-18T16:09:04Z" level=error msg="Failed to execute kubectl apply: error: error validating \"https://cloud.weave.works/k8s.yaml?k8s-version=v1.12.1&t=rfc9saahqrj9xnap6cyyyjsqjxp9e4mj&omit-support-inf
o=true\": error validating data: [field spec.template.spec.containers[0].securityContext.procMount for v1.SecurityContext is required, field spec.template.spec.containers[0].securityContext.procMount for v1.Sec
urityContext is required]; if you choose to ignore these errors, turn validation off with --validate=false\nFull output:\nerror: error validating \"https://cloud.weave.works/k8s.yaml?k8s-version=v1.12.1&t=rfc9s
aahqrj9xnap6cyyyjsqjxp9e4mj&omit-support-info=true\": error validating data: [field spec.template.spec.containers[0].securityContext.procMount for v1.SecurityContext is required, field spec.template.spec.contai
ners[0].securityContext.procMount for v1.SecurityContext is required]; if you choose to ignore these errors, turn validation off with --validate=false"
```

We're seemingly missing a required security context field that would grant our agents access to the proc filesystem.

The cluster in question doesn't seem to have any security policies visible to the user:

```
$ kubectl get psp
No resources found.
```

**Comment by rade** [comment]
This is a bug in kubernetes 1.12. See https://github.com/kubernetes/kubernetes/issues/69647. Should be fixed in the next patch release.

**Comment by dholbach** [comment]
In an [intercom discussion](https://app.intercom.io/a/apps/nvwp88sj/inbox/inbox/1843527/conversations/19084001799), a user suggested to include a default value in our yaml.

Not quite sure what they meant, but is this an option?

**Comment by alexellis** [comment]
Hi @dholbach what's the fix? (stay on 1.11?)

**Comment by rade** [comment]
1.12.2 should have the fix
